src/store/middleware/model-poller.ts

```diff
@@ -138,13 +145,29 @@ export const modelPollerMiddleware: Middleware = (reduxStore) => {
               }, 30000);
             });
           } while (isLoggedIn(wsControllerURL, reduxStore.getState()));
-        });
-        return;
-      } else if (action.type === actionsList.logOut) {
-        jujus.forEach((juju) => {
-          juju.logout();
-        });
-      }
+        }
+      );
+      return;
+    } else if (action.type === actionsList.logOut) {
+      jujus.forEach((juju) => {
+        juju.logout();
+      });
       return next(action);
-    };
+    } else if (action.type === actionsList.updatePermissions) {
```

Comment:
This now needs to be handled in the middleware as it need to have access to the controller connections which are only stored in the middleware (previously they were stored in the redux store).